### PR TITLE
AYS-345 | Unable to Delete Roles Not Associated with Institutions

### PR DESCRIPTION
**Pull request by MenekseYuncu**
### Checklist

Before submitting your pull request, ensure the following:

- [x] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#branch-naming-conventions).

- [x] **Local Testing:**
    - I have tested my changes locally on Postman, and they are working as expected.

- [x] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [x] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes made in Postman.

- [x] **Testing:**
    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [x] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [x] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

### CODE REVIEW

File: src/test/java/org/ays/auth/service/impl/AysRoleUpdateServiceImplTest.java [hunk 1]
```diff
@@ -740,6 +760,43 @@ void givenValidId_whenRoleUsing_thenThrowAysRoleAssignedToUserException() {
...
+        // When
+        AysRole mockRole = new AysRoleBuilder()
+                .withValidValues()
+                .withId(mockId)
+                .build();
+        Mockito.when(roleReadPort.findById(Mockito.anyString()))
+                .thenReturn(Optional.of(mockRole));
+
+        Mockito.when(identity.getInstitutionId())
+                .thenReturn(UUID.randomUUID().toString());
```
Comment (agitrubard) [inline src/test/java/org/ays/auth/service/impl/AysRoleUpdateServiceImplTest.java:785]: Burada da yukarıda olduğu gibi yeni bir ID oluşturarak kullanabiliriz

**Review by agitrubard — CHANGES_REQUESTED** [review]

File: src/test/java/org/ays/auth/service/impl/AysRoleUpdateServiceImplTest.java (hunk 1 above)
Comment (MenekseYuncu) [reply src/test/java/org/ays/auth/service/impl/AysRoleUpdateServiceImplTest.java:785]: String mockId = "32657ecb-5172-401d-8b29-a8e4454e8243"; bunun gibi mi?

Comment (agitrubard) [reply src/test/java/org/ays/auth/service/impl/AysRoleUpdateServiceImplTest.java:785]: Evet bu şekilde tanımlayabiliriz 👍🏼 fakat aynı ID olmamalı, çünkü testte farklı bir ID olacak diyoruz

File: src/test/java/org/ays/auth/service/impl/AysRoleUpdateServiceImplTest.java [hunk 2]
```diff
@@ -740,6 +760,44 @@ void givenValidId_whenRoleUsing_thenThrowAysRoleAssignedToUserException() {
...
+        // When
+        AysRole mockRole = new AysRoleBuilder()
+                .withValidValues()
+                .withId(mockId)
+                .build();
+        Mockito.when(roleReadPort.findById(Mockito.anyString()))
+                .thenReturn(Optional.of(mockRole));
+
+        Mockito.when(identity.getInstitutionId())
+                .thenReturn(mockRandomInstitutionId);
```
Comment (agitrubard) [inline src/test/java/org/ays/auth/service/impl/AysRoleUpdateServiceImplTest.java:786]: Bu bilgi dışarıdan gelmediği için `Given` alanına yazmamalıyız
```suggestion

        // When
        AysRole mockRole = new AysRoleBuilder()
                .withValidValues()
                .withId(mockId)
                .build();
        Mockito.when(roleReadPort.findById(Mockito.anyString()))
                .thenReturn(Optional.of(mockRole));

        Mockito.when(identity.getInstitutionId())
                .thenReturn("3b9dcab2-028f-4bb8-9c9b-23130d9579ae");
```

**Review by agitrubard — CHANGES_REQUESTED** [review]

**Review by agitrubard — APPROVED** [review]

**Review by egehanasal — APPROVED** [review]

**Pull request merged**